src/scripts/main.js

```diff
@@ -1,3 +1,175 @@
+  const cell = Math.floor(Math.random() * size);
+
+  (field[row][cell] === 0)
+    ? field[row][cell] = Math.random() > 0.1 ? 2 : 4
+    : fillRandomCell();
+}
+
+function fillGameField() {
+  for (let row = 0; row < size; row++) {
+    for (let cell = 0; cell < size; cell++) {
+      const currentRenderCell = fieldRows[row].children[cell];
+      const currentCell = field[row][cell];
+
+      currentRenderCell.innerText = currentCell || '';
+
+      currentRenderCell.className = currentCell
+        ? `field-cell field-cell--${currentCell}`
+        : 'field-cell';
+
+      if (currentCell === 2048) {
+        document.removeEventListener('keydown', handleArrows);
+        winMessage.classList.remove('hidden');
+      }
+    }
+  }
+
+  gameScore.innerText = score;
+  checkGameOver();
+}
+
+function checkGameOver() {
+  let gameOver = true;
+  const transposedField = transpose([...field]);
+  const fieldContainsZero = field.some(row => row.includes(0));
+
+  for (let row = 0; row < size; row++) {
+    for (let cell = 0; cell < size - 1; cell++) {
+      if (field[row][cell] === field[row][cell + 1]
+        || transposedField[row][cell] === transposedField[row][cell + 1]) {
+        gameOver = false;
+      }
+    }
+  }
+
+  if (gameOver && !fieldContainsZero) {
+    document.removeEventListener('keydown', handleArrows);
+    loseMessage.classList.remove('hidden');
+  }
+}
+
+function transpose(arr) {
+  return arr.map((_, columnIndex) => arr.map(row => row[columnIndex]));
+}
+
+function mergeRowCells(row) {
+  for (let i = 0; i < row.length - 1; i++) {
+    if (row[i] === row[i + 1]) {
+      row[i] *= 2;
+      row.splice(i + 1, 1);
+      score += row[i];
+    }
+  }
+
+  while (row.length < size) {
+    row.push(0);
+  }
+
+  return row;
+}
+
+function removeZeros(arr) {
+  return arr.filter(value => value !== 0);
+}
+
+function clickArrowLeft() {
+  for (let row = 0; row < size; row++) {
+    const filledCells = removeZeros(field[row]);
+
+    mergeRowCells(filledCells);
+    field[row] = filledCells;
+  }
+}
+
+function clickArrowRight() {
+  for (let row = 0; row < size; row++) {
+    const filledCells = removeZeros(field[row].reverse());
+
+    mergeRowCells(filledCells);
+    field[row] = filledCells.reverse();
+  }
+}
+
+function clickArrowUp() {
+  field = transpose(field);
+  clickArrowLeft();
+  field = transpose(field);
+}
+
+function clickArrowDown() {
+  field = transpose(field);
+  clickArrowRight();
+  field = transpose(field);
+}
+
+function handleArrows(e) {
+  switch (e.key) {
+    case 'ArrowLeft':
+      clickArrowLeft();
+      fillRandomCell();
+      fillGameField();
+      break;
+
+    case 'ArrowRight':
+      clickArrowRight();
+      fillRandomCell();
+      fillGameField();
+      break;
+
+    case 'ArrowUp':
+      clickArrowUp();
+      fillRandomCell();
+      fillGameField();
+      break;
+
+    case 'ArrowDown':
+      clickArrowDown();
+      fillRandomCell();
+      fillGameField();
```

Comment:
don't repeat yourself
create separate functions to avoid repetitions (`fillRandomCell(), fillGameField()`)